packages/ia-components/sandbox/audio-players/audio-player-main.jsx

```diff
@@ -72,16 +73,31 @@ export default class TheatreAudioPlayer extends Component {
    */
   showMedia() {
     const { source, sourceData } = this.props;
-    const isExternal = source === 'youtube' || source === 'spotify';
     let mediaElement = (
       <ArchiveAudioPlayer
         {...this.props}
         onRegistrationComplete={this.receiveURLSetter}
         needsURLSettingAccess
       />
-    );
-    if (isExternal) {
-      // make iframe with URL
+    ); // shorten code
+    if (source === 'youtube') {
+      const { urlSetterFN } = this.state;
+      const externalSourceDetails = sourceData[source] || {};
```

Comment:
This variable instantiation is identical in both conditions, and can therefore be done prior to the conditional blocks.